Dev

Stellareign opened the pull request.

Stellareign committed:
Немного поправок

Rumonz1 committed:
Исправил код в соответствии с замечаниями наставника

Stellareign committed:
Корректировка по замечаниям.

Stellareign committed:
Merge remote-tracking branch 'origin/dev' into dev

Stellareign committed:
Merge branch 'RuAna-4' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/service/impl/AuthServiceImpl.java

Stellareign committed:
Merge remote-tracking branch 'origin/Timur2.1revork' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/CommentsController.java
#	src/main/java/ru/skypro/homework/service/impl/CommentServiceImpl.java
#	src/main/java/ru/skypro/homework/service/impl/MapperUtil/CommentMapping.java

Stellareign committed:
Совместные правки

Stellareign committed:
Совместные правки

Stellareign committed:
Совместные правки

Stellareign committed:
Немного поправок

Stellareign committed:
Немного поправок

alkor63 committed:
Компилируется, но работает не корректно
так ине смог победить "Content type 'application/octet-stream' not supported"

alkor63 committed:
Компилируется, но работает не корректно
так ине смог победить "Content type 'application/octet-stream' not supported"
при добавлении нового объявления
При редактировании картинка ошибка не выскакивает

alkor63 committed:
Добавил аннотацию @Component и чудо свершилось!
Объявление из Фронта попало в БД

Stellareign committed:
Чуток подправлены комменты. Создание коммента работает, остальное  - доработать.

Stellareign committed:
Чуток подправлены комменты. Создание коммента работает, остальное  - доработать.

Stellareign committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/dto/ads/Ad.java
#	src/main/java/ru/skypro/homework/dto/ads/CreateOrUpdateAd.java
#	src/main/java/ru/skypro/homework/service/MapperUtil/MapperUtilAds.java

Stellareign committed:
Merge remote-tracking branch 'origin/dev' into dev

Stellareign committed:
Поборола интейфейс

Stellareign committed:
Вернула для Ad то, что случайно удалила

Stellareign committed:
Вернула для Ad то, что случайно удалила

Stellareign committed:
Немного поправок

Stellareign committed:
Немного поправок

Stellareign committed:
Немного поправок

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/ExtendedAdDTO.java:
```diff
@@ -2,16 +2,20 @@
 @AllArgsConstructor
+@NoArgsConstructor
 @Data
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/CreateOrUpdateAd.java:
```diff
@@ -2,9 +2,14 @@
-public class CreateOrUpdateAdDTO {
+@NoArgsConstructor
+@AllArgsConstructor
+@Component
```
зачем здесь аннотация Component?

alkor63 replied:
Дима, подскажи, пожалуйста, как победить ошибку 415 - Content type 'application/octet-stream' not supported ?

alkor63 replied:
... при попытке создать объявление

dmitry-bizin replied:
никак её не надо побеждать, это ошибка Swagger UI, что он не умеет отправлять правильно такие запросы. Я не знаю, говорили вам или нет, но вот сейчас говорю - никогда не используйте Swagger UI, потому что он дико багнутый. Надо использовать Postman или идеевский http-клиент.

dmitry-bizin replied:
По поводу аннотации Component и также прочих аннотаций сильно рекомендую вернуться на второй курс и пересмотреть там уроки по Spring, потому что это проблема, что вы до сих пор не понимаете, для чего эта аннотация и как слепые котята тыкаетесь, перебирая разные варианты.

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java:
```diff
@@ -11,7 +12,8 @@
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/Ad.java:
```diff
@@ -1,8 +1,14 @@
+import org.springframework.stereotype.Component;
 
 @Data
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/authorization/Register.java:
```diff
@@ -1,9 +1,11 @@
 import ru.skypro.homework.dto.Role;
 
 @Data
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CommentDTO.java:
```diff
@@ -4,18 +4,12 @@
 @AllArgsConstructor
 @NoArgsConstructor
-
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CommentsDTO.java:
```diff
@@ -15,6 +15,7 @@
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CreateOrUpdateCommentDTO.java:
```diff
@@ -3,10 +3,12 @@
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
+@Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/AuthUserDTO.java:
```diff
@@ -0,0 +1,20 @@
+import ru.skypro.homework.dto.Role;
+
+@Data
+@Component
```
зачем здесь аннотация Component?

BizinMitya commented on src/main/java/ru/skypro/homework/dto/user/AuthUserDTO.java:
```diff
@@ -0,0 +1,20 @@
+@Component
+@AllArgsConstructor
+@NoArgsConstructor
+public class AuthUserDTO {
```
Такой dto нет в спецификации

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/UserDTO.java:
```diff
@@ -5,15 +5,13 @@
-@Setter
 @AllArgsConstructor
 @NoArgsConstructor
 @Component
```
зачем здесь аннотация Component?

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/MapperUtil/CommentMapping.java:
```diff
@@ -17,21 +18,20 @@ public CommentDTO mapToDto(Comment entity) {
         entity.setAuthor(author);
         entity.setAdId(ad);
-        entity.setCreatedAt(LocalDateTime.from(Instant.now()));
+        entity.setCreatedAt(entity.getCreatedAt()); //<-- не могу понять, нужно ли здесь это???
```
нет, потому что кладётся то, что достаётся через геттер - это бессмысленно. Надо присваиваить текущее время - время создания как раз

Rumonz1 replied:
Ага...получается сначала сделал правильно

dmitry-bizin replied:
Да, было правильно

Rumonz1 replied:
Дмитрий, так ведь нам нужен лонг а не LDT , как тогда было правильно?

dmitry-bizin replied:
И? Какая разница, что нужно фронту, если нам выгоднее и удобнее работать с LDT. Ещё раз, уже в который, повторю: DTO и сущности - это РАЗНЫЕ вещи, там могут быть РАЗНЫЕ данные, в РАЗНОМ порядке и в РАЗНОМ наборе, но есть такая штука, как МАППИНГ - то, чем предстоит заниматься в следующей домашке. Как раз при помощи МАППИНГА и надо будет настроить мост между фронтом и бэком.

BizinMitya commented on src/main/java/ru/skypro/homework/service/MapperUtil/MapperUtilAds.java:
```diff
@@ -7,21 +7,24 @@
 
 @Data
 @NoArgsConstructor
 @Configuration
```
```suggestion
@Component
```

Rumonz1 committed:
В эфире программа ээээксперименты

alkor63 committed:
Все методы контроллера/сервиса вроде работают нормально: объявления можно посмотреть, добавить, отредактировать (в том числе картинку), удалить
Картинки на диск сохраняются с заданным именем, но на фронте не отображаются.
Проверка авторизации пока тоже не работает :(

Stellareign committed:
Вернула для Ad то, что случайно удалила

Stellareign committed:
Merge branch 'RuAna-5' into dev

Stellareign committed:
Merge branch 'aleksey6' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/dto/Role.java

Stellareign committed:
НЕмного работы с картинками.
Осталось разобраться, как их выгрузить на фронт

Stellareign committed:
Маленькая правка в  UserDTOFactoryImpl

Stellareign committed:
Переделала полностью хранение картинок на хранение в папке
Удалила классы Avatar, AvatarRepository.java
Получение аватара на фронт пока так и не удалось победить

Stellareign committed:
Переделала полностью хранение картинок на хранение в папке
Удалила классы Avatar, AvatarRepository.java
Получение аватара на фронт пока так и не удалось победить

Stellareign committed:
немного правок

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -21,14 +22,15 @@ public class WebSecurityConfig {
             "/login",
             "/register",
-            "/ads/*/image",
+            "/ads/image",
             "/users/*/image"
```
зачем здесь это?

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -38,17 +40,17 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                                         .mvcMatchers(AUTH_WHITELIST)
                                         .permitAll()
-                                        .mvcMatchers(HttpMethod.GET, "/ads", "/image/**")
+                                        .mvcMatchers(HttpMethod.GET, "/ads","/ads/me", "/users/me")
```
зачем здесь /ads/me и /users/me? это неверно, эти методы должны быть под аутентификацией, а сейчас они у вас открыты. Сильно рекомендую вернуть в этом классе всё как было вам дано и не ломать в нём ничего до 4ой домашки.

BizinMitya commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -62,94 +76,67 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
-    @PreAuthorize("hasRole('USER') and @adsService.getAdById(#adId).email == authentication.principal.username")
-
+    @Operation(summary = "Получение информации об объявлении по id")
+    @PreAuthorize("hasRole('ROLE_USER') and @adsService.getAdById(#adId).email == authentication.principal.username")
```
это неверно, должно быть разрешено админу, а не юзеру

dmitry-bizin replied:
это неверно, метод должен быть разрешён всем юзерам

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -62,94 +76,67 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
-//            "@adsService.getAdById(#adId).email == authentication.principal.username)")
-
+    @Operation(summary = "Удалить объявление по id")
+    @PreAuthorize("hasRole('ADMIN') or (hasRole('USER') and" +
```
на юзера проверять нет смысла

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -62,94 +76,67 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
-
-    @Operation(summary = "Обновление информации об объявлении")
+    @Operation(summary = "Обновить объявление по id")
+    @PreAuthorize("hasRole('ADMIN') or (hasRole('USER') and" +
```
на юзера проверять нет смысла

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -62,94 +76,67 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
-    )
-//    @PreAuthorize("hasRole('USER')")
+    @Operation( summary = "Получение всех объявлений авторизованного пользователя")
+    @PreAuthorize("hasRole('USER')")
```
это неверно, метод должен быть разрешён всем юзерам

alkor63 replied:
Дима, привет! Здесь я тебя "недопонял". 
Метод, который показывает все объявления любому желающему был выше (getAdsDTO()), а это я понимал как "показать все МОИ объявления. 
Ты имеешь ввиду, что Админ тоже в этом смысле Юзер?

dmitry-bizin replied:
Да, все мои объявления, но у меня может быть роль ADMIN и я по-прежнему должен иметь возможность видеть свои объявления, но сейчас это запрещено.

dmitry-bizin commented on src/main/java/ru/skypro/homework/entity/Comment.java:
```diff
@@ -31,5 +31,11 @@ public class Comment {
     @JoinColumn(name = "user_id")
     private User author; // id автора комментария
 
+    public void convertTime(){
+        Date  date = new Date(createdAt);
+        Format format = new SimpleDateFormat("dd MMM yyyy HH:mm:ss:SSS");
+        format.format(date);
+    }
```
Зачем этот метод? какой в нём смысл?

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/MapperUtil/CommentMapping.java:
```diff
@@ -31,8 +32,10 @@ public Comment mapToEntity(CreateOrUpdateCommentDTO createOrUpdateComment, User
         entity.setAuthor(author);
         entity.setAdId(ad);
-        entity.setCreatedAt(entity.getCreatedAt()); //<-- не могу понять, нужно ли здесь это???
+        entity.convertTime(); //<- нашёл в интернете способ конвертирования лонга в дату
```
зачем здесь что-то конвертировать-то? С фронта НЕ ПРИХОДИТ даты и время, с фронта приходит ТОЛЬКО ТЕКСТ комментария, поэтому всё остальное НАДО СЕТИТЬ самим, включая автора и дату и время создания комментария (просто надо взять текущую дату и время, что сложного?)

Rumonz1 committed:
внёс поправку в маппере

Stellareign committed:
1

Stellareign committed:
Merge remote-tracking branch 'origin/RuAna-6' into RuAna-6

Stellareign committed:
1

Stellareign committed:
Merge branch 'dev' into RuAna-6

Stellareign committed:
Починила отображение комментов: заменила метод репозитория по вытаскиванию листа.

Stellareign committed:
Починила отображение комментов: заменила метод репозитория по вытаскиванию листа.

Stellareign committed:
просто переименовала ва метода по комментам

Stellareign committed:
WebSecurityConfig

Stellareign committed:
WebSecurityConfig

alkor63 committed:
Учел замечания Наставника

alkor63 committed:
Учел замечания Наставника

alkor63 committed:
Merge branch 'aleksey6' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/dto/authorization/Register.java
#	src/main/java/ru/skypro/homework/service/MapperUtil/CommentMapping.java

Stellareign committed:
Удалила две строки из WebScurity

Stellareign committed:
Удалила две строки из WebScurity

Stellareign committed:
Merge branch 'dev' into RuAna-6

# Conflicts:
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java

Stellareign committed:
По комментам пара правок

Stellareign committed:
Исправление после слияния.

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -77,7 +77,7 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
     // получение информации об объявлении
     @Operation(summary = "Получение информации об объявлении по id")
-    @PreAuthorize("hasRole('ROLE_USER') and @adsService.getAdById(#adId).email == authentication.principal.username")
+    @PreAuthorize("hasRole('ROLE_USER')")
```
сейчас запрещено админу, а должно быть разрешено всем - надо удалить эту аннотацию

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -119,7 +119,7 @@ public ResponseEntity<Ad> updateAd(@PathVariable int adId, @RequestBody CreateOr
 
     @Operation( summary = "Получение всех объявлений авторизованного пользователя")
-    @PreAuthorize("hasRole('USER')")
+    @PreAuthorize("hasRole('ADMIN') or hasRole('USER')")
```
если удалить аннотацию, то смысл не поменяется, так зачем её писать?

alkor63 replied:
я рассуждал: есть Роль - значит авторизован
Без аннотации запрос сможет делать неавторизованный пользователь

dmitry-bizin replied:
Не сможет - посмотри в WebSecurityConfig внимательнее (если вы его снова не сломали только)

alkor63 committed:
Учел замечания наставника - убрал 2 строчки в AdsController
 Залез на огород Тимура - внёс правки в CommentController/service/repository
 теперь оно работает.
 Каскадное удаление комментов вместе с объявлением пока не победил (

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -18,53 +18,42 @@
-        return ResponseEntity.notFound().build();
-        }
-
+    public void deleteComment(@PathVariable("commentId") int pk ) {
```
где параметр adId?

Stellareign committed:
Вернула адрес удаления и обновления комментария

Stellareign committed:
Вернула адрес удаления и обновления комментария

Stellareign committed:
Всё работает, кроме картинок

Stellareign committed:
Всё работает, кроме картинок

alkor63 committed:
Еще несколько мелких правок

Stellareign committed:
Merge branch 'RuAna-6' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/repository/CommentRepository.java
#	src/main/java/ru/skypro/homework/service/impl/CommentServiceImpl.java
#	src/main/java/ru/skypro/homework/service/interfaces/CommentsService.java
#	src/main/resources/application.properties

Rumonz1 committed:
Проверил работу контроллеров комментов в сваггере, работает хорошо. +Решил попробовать настроить добавление фото объявлений

alkor63 committed:
Merge remote-tracking branch 'origin/dev' into dev

alkor63 committed:
Подтянул изменения Насти

Stellareign committed:
Не работает обновление и удаление комментов с фильтрацией

Stellareign committed:
Не работает обновление и удаление комментов с фильтрацией

Stellareign committed:
Не работает обновление и удаление комментов с фильтрацией

Stellareign committed:
Вроде бы есть разграничения по ролям!

Stellareign committed:
Не работает аутентификация :(

Stellareign committed:
Не работает аутентификация :(

Stellareign committed:
Не работает аутентификация :(

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -24,30 +31,41 @@ public class WebSecurityConfig {
                                 authorization
                                         .mvcMatchers(AUTH_WHITELIST)
                                         .permitAll()
+                                        .mvcMatchers(HttpMethod.GET, "ads", "/image")
```
```suggestion
                                        .mvcMatchers(HttpMethod.GET, "/ads")
```

dmitry-bizin replied:
картинок пока нет, поэтому на будущее не нужно добавлять неизвестно что

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -24,30 +31,41 @@ public class WebSecurityConfig {
+     * Конфигурация для фильтра цепочки безопасности (SecurityFilterChain) в Spring Security.
+     * Он определяет правила доступа для различных URL-адресов и HTTP-методов.
+     */
+    @Autowired
+    MyUserDetails myUserDetails;
```
это не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -83,6 +85,7 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
     @ApiResponse(responseCode = "401", description = "Unauthorized"),
     @ApiResponse(responseCode = "404", description = "Not Found")
 })
+    @PreAuthorize("hasRole('ADMIN') or " + "hasAuthority(authentication.getName())")
```
это неверно, эндпоинт должен доступен всем ролям, аннотацию надо удалить

alkor63 replied:
А неавторизованный пользователь? Или он сюда через фронт не доберётся?

dmitry-bizin replied:
есть аутентификация, а есть авторизация - это разные вещи. Так вот аутентификация у вас настроена в WebSecurityConfig, а здесь вы настраиваете авторизацию - право на вызов эндпоинта админу или создателю

dmitry-bizin commented on src/main/java/ru/skypro/homework/security_service/MyUserDetails.java:
```diff
@@ -25,16 +28,23 @@ public String getUsername() {
+                .map(role -> "ROLE_" + role)
+                .map(SimpleGrantedAuthority::new)
+                .map(Collections::singleton)
+                .orElse(Collections.emptySet());
```
```suggestion
                .orElseGet(Collections::emptySet);
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -118,17 +124,17 @@ public ResponseEntity<Ad> updateAd(@PathVariable int adId, @RequestBody CreateOr
+
     @Operation(summary = "Обновление картинки объявления")
-    @PreAuthorize("hasRole('ADMIN') and @adsService.getAdById(#adId).email == authentication.principal.username")
+    @PreAuthorize("@adsService.getAdById(#adId).email == authentication.principal.username")
```
```suggestion
    @PreAuthorize("hasRole('ADMIN') or @adsService.getAdById(#adId).email == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -91,14 +94,17 @@ public ResponseEntity<ExtendedAdDTO> getAdById(@PathVariable int adId) {
     @Operation(summary = "Удалить объявление по id")
     @PreAuthorize("hasRole('ADMIN') or " +
             "@adsService.getAdById(#adId).email == authentication.principal.username")
```
```suggestion
            "@adsService.getAdById(#adId).email == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -91,14 +94,17 @@ public ResponseEntity<ExtendedAdDTO> getAdById(@PathVariable int adId) {
+
                 adsService.deleteAdsById(adId);
     }
 
```
```
@PreAuthorize("hasRole('ADMIN') or " +
            "@adsService.getAdById(#adId).email == authentication.name")
```

dmitry-bizin replied:
для обновления

Stellareign committed:
Правки по замечаниям

Stellareign committed:
Правки по замечаниям

Stellareign committed:
Правки по замечаниям

Stellareign committed:
Правки по замечаниям

alkor63 committed:
Merge remote-tracking branch 'origin/dev' into dev

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -85,15 +84,14 @@ public ResponseEntity<Ad> addAd(@RequestPart("image") MultipartFile image,
     @Operation(summary = "Удалить объявление по id")
-    @PreAuthorize("hasRole('ADMIN') or " +
-            "@adsService.getAdById(#adId).email == authentication.principal.username")
+    @PreAuthorize("hasRole('ROLE_ADMIN') or " +
```
```suggestion
    @PreAuthorize("hasRole('ADMIN') or " +
```
или
```suggestion
    @PreAuthorize("hasAuthority('ROLE_ADMIN') or " +
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -109,8 +107,8 @@ public void removeAdById(@PathVariable int adId) throws IOException {
     @Operation(summary = "Обновить объявление по id")
-    @PreAuthorize("hasRole('ADMIN') or " +
-            "@adsService.getAdById(#adId).email == authentication.principal.username")
+    @PreAuthorize("hasRole('ROLE_ADMIN') or " +
```
```suggestion
    @PreAuthorize("hasRole('ADMIN') or " +
```
или
```suggestion
    @PreAuthorize("hasAuthority('ROLE_ADMIN') or " +
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -134,7 +132,7 @@ public ResponseEntity<AdsDTO> getCurrentUserAds(Authentication authentication) {
 
     @Operation(summary = "Обновление картинки объявления")
-    @PreAuthorize("@adsService.getAdById(#adId).email == authentication.principal.username")
+    @PreAuthorize("hasRole('ROLE_ADMIN') or @adsService.getAdById(#adId).email == authentication.name")
```
```suggestion
    @PreAuthorize("hasRole('ADMIN') or @adsService.getAdById(#adId).email == authentication.name")
```
или
```suggestion
    @PreAuthorize("hasAuthority('ROLE_ADMIN') or @adsService.getAdById(#adId).email == authentication.name")
```

alkor63 committed:
Учёл замечания Наставника:
несколько правок в AdsController,
WebSecurityConfig править не пришлось
MyUserDetails - за настей, у меня такого фрагмента нет

alkor63 committed:
Merge branch 'Alexey' into dev

# Conflicts:
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java

Stellareign committed:
Преавторизация всё равно не работает

Stellareign committed:
Преавторизация всё равно не работает

Stellareign committed:
Измени проверку методов контроллера на доступ

Stellareign committed:
Измени проверку методов контроллера на доступ

Rumonz1 committed:
Merge branch 'dev' into Timur4.0

Rumonz1 committed:
добавил проверку аунтефикации в контроллер комментов

Stellareign committed:
исправила проверку в adsEdit

Stellareign committed:
исправила проверку в public AdEntity updateImage`

Stellareign committed:
Добавила проверки на роли в виде методов checkAccessToComments() & checkAccessToAd()

Stellareign committed:
Добавила проверки на роли в виде методов checkAccessToComments() & checkAccessToAd()

Stellareign committed:
Добавила проверки на роли в виде методов checkAccessToComments() & checkAccessToAd()

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -14,158 +17,146 @@
+            @ApiResponse(responseCode = "404", description = "Not Found")
+    })
+    @PreAuthorize("hasRole('ADMIN') or " +
+            "@adsService.getAdById(#adId).email == authentication.name")
```
```suggestion
            "@adsRepository.findByPk(#adId).author.username == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -14,158 +17,146 @@
-
+    @Operation(summary = "Обновить объявление по id")
+    @PreAuthorize("hasRole('ADMIN') or " +
+            "@adsService.getAdById(#adId).email == authentication.name")
```
```suggestion
            "@adsRepository.findByPk(#adId).author.username == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -14,158 +17,146 @@
-    @PatchMapping(value = "/{adId}/image", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
 
+    @Operation(summary = "Обновление картинки объявления")
+    @PreAuthorize("hasRole('ADMIN') and @adsService.getAdById(#adId).email == authentication.name")
```
```suggestion
    @PreAuthorize("hasRole('ADMIN') and @adsRepository.findByPk(#adId).author.username == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -29,69 +25,51 @@
+
     // удаление комментария по id
     @Operation(summary = "Удаление комментария")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN') or @commentsService.getAuthorByCommentId(#pk).username == authentication.principal.username")
```
```suggestion
    @PreAuthorize("hasAuthority('ROLE_ADMIN') or @commentsService.getAuthorByCommentId(#pk).username == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -29,69 +25,51 @@
-          }
-
+    @Operation(summary = "Обновление комментария")
+    @PreAuthorize("hasAuthority('ROLE_ADMIN') or @commentsService.getAuthorByCommentId(#pk).username == authentication.principal.username")
```
```suggestion
    @PreAuthorize("hasAuthority('ROLE_ADMIN') or @commentsService.getAuthorByCommentId(#pk).username == authentication.name")
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/CommentServiceImpl.java:
```diff
@@ -31,42 +36,66 @@ public class CommentServiceImpl implements CommentsService {
-    public void deleteComment(Integer adId, Integer pk) {
-        commentRepository.deleteByPkAndAdId(pk, adId);
+    public User getAuthorByCommentId(Integer pk) {
+        if (commentRepository.findById(pk).isPresent()) {
+            return commentRepository.findById(pk).get().getAuthor();
```
Зачем 2 раза вызывать один и тот же метод? надо ОДИН раз вызвать, получить результат и далее работать с результатом, а не вызывать 10 раз метод

Stellareign committed:
Исправила по замечаниям. Топтались вокруг да около. Теперь, вроде, всё заработало.
